Training/Training/Reversal.c

```diff
@@ -0,0 +1,181 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment4 ~ A training program for new joiners at Metamation, Batch - Sept 2024.
+// Copyright (c) Gokul TS @Metamation India.
+// ------------------------------------------------------------------
+// Reversal.c
+// C program to Check Palindrome or not
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#define ANSI_RESET_ALL "\x1b[0m"
+#define ANSI_COLOR_RED "\x1b[31m"
+#define ANSI_COLOR_GREEN "\x1b[32m"
+#define ANSI_COLOR_BLUE "\x1b[36m"
+#define ANSI_COLOR_MAGENTA "\x1b[35m"
+
+#define MAX_DIGIT 13
+#define MAX_CHAR 50
+
+/// <summary>Function to check given number is palindrome or not </summary>
+bool NumIsPalindrome (int num);
+/// <summary>Function to check given string is palindrome or not </summary>
+bool StrIsPalindrome (char* phrase);
+/// <summary>Function to get integer input from user</summary>
+void IntegerInput (int option);
+/// <summary>Function to get string input from user</summary>
+void StringInput (int option);
+/// <summary>Function to check test cases</summary>
+void OutputCheck ();
+/// <summary>Function to print result</summary>
+void PrintResult (bool value);
+/// <summary>Clears the content on screen </summary>
+void ClrScreen ();
+/// <summary>Clears buffer values </summary>
+void ClrBuffer (char option);
+
+
+void main () {
+   while (1) {
+      bool breakLoop = false;
+      int count = 0;
+      printf (ANSI_COLOR_MAGENTA"1-PALINDROME CHECKER\n2-REVERSE A NUMBER\n3-TEST CASES\n4-CLEAR SCREEN\n5-EXIT\n"ANSI_RESET_ALL"Enter an option:");
+      char option = fgetc (stdin);
+      int num = option;
+      if (num != 10) {
+         for (;;) {
+            option = fgetc (stdin);
+            if (option == '\n') break;
+            count++;
+         }
+      }
+      if (count == 0 && (num < 54 && num > 48)) {
+         switch (num) {
+            case '1': {
+               StringInput (option);
+               break;
+            }
+            case '2': {
+               IntegerInput (option);
+               break;
+            }
+            case '3': {
+               OutputCheck ();
+               break;
+            }
+            case '4': {
+               ClrScreen ();
+               break;
+            }
+            case '5': {
+               breakLoop = true;

```

Comment:
Instead of this flag return 0
```suggestion
               breakLoop = true;
return 0;
```